vm/adb/adb.go

```diff
@@ -373,6 +381,32 @@ func (inst *instance) getBatteryLevel(numRetry int) (int, error) {
 	return val, nil
 }
 
+// This function was added to handled scenarios where the device being fuzzed
+// does not have a battery and is not running the battery service.
+// It first checks if the device is running a battery service.
+// If not, then the device is assumed to be battery-less (powered) and OK
+// to fuzz. The function returns without an error in that case. If the
+// device has a battery service, or a lack of service is not confirmed, then
+// the battery status is checked and the result returned.
+
+func (inst *instance) doOneBatteryCheck() ([]byte, error) {
+	out1, err1 := inst.adb("shell", "dumpsys battery | grep level:")
+	if err1 != nil {
+		// There is a possibility that the device does not have
+		// a battery. Check for that to allow such devices to be fuzzed.
+		out2, err2 := inst.adb("shell", "service check battery | grep \"not found\"")
+		if err2 == nil {
+			// the device executed the shell command but is reporting
+			// that the service was not found. Assume it is powered and
+			// OK to fuzz, as may be the case for an embedded device.
+			// Returning "non-battery device" in error string to indicate
+			// that.
+			return out2, fmt.Errorf("non-battery device")
```

Comment:
Just return 100% battery level, if there is no battery. This will make caller much simpler:
```
return []byte("level: 100"), nil
```